### PR TITLE
packaging: Configure QEMU with --enable-pie

### PR DESCRIPTION
**Pull request by gkurz**
We explicitely set the Postion Independant Executlable (PIE) options
in the extra CFLAGS and LDFLAGS that are passed to the QEMU configure
script for all archs. This means that these options are used pretty
much everywhere, including when building the sample plugins under the
test directory. These cannot be linked with -pie and break the build,
as experienced recently on ARM (see PR #2732).

This only broke on ARM because other archs are configured with
--disable-tcg : this disables plugins which are built by default
otherwise.

The --enable-pie option is all that is needed. The QEMU build system
knows which binaries should be created as PIE, e.g. the important
bits like QEMU and virtiofsd, and which ones should not, e.g. the
sample plugins that aren't used in production.

Rely on --enable-pie only, for all archs. This allows to drop the
workaround that was put in place in PR #2732.

Fixes: #2757
Signed-off-by: Greg Kurz <groug@kaod.org>

### CODE REVIEW

File: tools/packaging/scripts/configure-hypervisor.sh [hunk 1]
```diff
@@ -434,12 +434,15 @@ generate_qemu_options() {
...
-		[ "${qemu_version}" == "6.1.0" ] && qemu_options+=(arch:"--disable-plugins")
-	fi
+	# SECURITY: Create binary as a Position Independant Executable,
+	# and take advantage of ASLR, making ROP attacks much harder to perform.
+	# (https://wiki.debian.org/Hardening)
+	case "$arch" in
+	aarch64) [ -z "${static}" ] && qemu_options+=(arch:"--enable-pie") ;;
+	x86_64) [ -z "${static}" ] && qemu_options+=(arch:"--enable-pie") ;;
+	ppc64le) [ -z "${static}" ] && qemu_options+=(arch:"--enable-pie") ;;
+	s390x) [ -z "${static}" ] && qemu_options+=(arch:"--enable-pie") ;;
```
Comment (fidencio) [inline tools/packaging/scripts/configure-hypervisor.sh:444]: As we're doing this for all supported arches, do we still need the switch?  If so, why?

Comment (dgibson) [reply tools/packaging/scripts/configure-hypervisor.sh:444]: Seconded.  Apart from that LGTM.

**Review by devimc — APPROVED** [review]
thanks @gkurz

**Review by dgibson — APPROVED** [review]

**Comment by gkurz** [comment]
The failing ARM and s390x CIs are caused by https://github.com/kata-containers/tests/issues/4044 IIUC. This should be fixed with https://github.com/kata-containers/tests/pull/4048.

**Comment by snir911** [comment]
> The failing ARM and s390x CIs are caused by [kata-containers/tests#4044](https://github.com/kata-containers/tests/issues/4044) IIUC. This should be fixed with [kata-containers/tests#4048](https://github.com/kata-containers/tests/pull/4048).

Merged, let's see
/retest-arm
/retest-s390x

**Comment by gkurz** [comment]
> > The failing ARM and s390x CIs are caused by [kata-containers/tests#4044](https://github.com/kata-containers/tests/issues/4044) IIUC. This should be fixed with [kata-containers/tests#4048](https://github.com/kata-containers/tests/pull/4048).
> 
> Merged, let's see /retest-arm /retest-s390x

Still failing but now it seems to be issue https://github.com/kata-containers/kata-containers/issues/2779 (PR https://github.com/kata-containers/kata-containers/pull/2780)

**Review by amshinde — APPROVED** [review]

**Pull request merged**